plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -802,7 +802,7 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
       fc::microseconds   max_trx_cpu_usage = max_trx_time_ms < 0 ? fc::microseconds::maximum() : fc::milliseconds(max_trx_time_ms);
 
       auto future = transaction_metadata::start_recover_keys(trx,
-                                                             _thread_pool.get_executor(),
+                                                             chain.get_thread_pool(),
```

Comment:
Will this change have any side effects on controller's thread pool operation?

What in release/5.0 cause the slowdown? In release/3.2 the same producer_plugin's thread pool was used and 31k TPS was achieved.

Reply:
For additional information. `post` is simply the `boost::post()` line. `exec` is complete execution of the contract including loading WASM and everything. Clearly these `post()` calls are killing performance.
```
 post calls: 10000, total: 162933us, avg: 16us
 exec calls: 10000, total: 167621us, avg: 16us
```